include/ozo/impl/async_request.h

```diff
@@ -232,44 +246,52 @@ struct async_get_result_op : boost::asio::coroutine {
                 }
             }
 
-            if (auto res = get_result(get_connection(ctx_))) {
-                const auto status = result_status(*res);
-                switch (status) {
-                    case PGRES_SINGLE_TUPLE:
-                        process_and_done(std::move(res));
-                        return;
-                    case PGRES_TUPLES_OK:
-                        process_and_done(std::move(res));
-                        consume_result(get_connection(ctx_));
-                        return;
-                    case PGRES_COMMAND_OK:
-                        done();
-                        consume_result(get_connection(ctx_));
-                        return;
-                    case PGRES_BAD_RESPONSE:
-                        done(error::result_status_bad_response);
-                        consume_result(get_connection(ctx_));
-                        return;
-                    case PGRES_EMPTY_QUERY:
-                        done(error::result_status_empty_query);
-                        consume_result(get_connection(ctx_));
-                        return;
-                    case PGRES_FATAL_ERROR:
-                        done(result_error(*res));
-                        consume_result(get_connection(ctx_));
-                        return;
-                    case PGRES_COPY_OUT:
-                    case PGRES_COPY_IN:
-                    case PGRES_COPY_BOTH:
-                    case PGRES_NONFATAL_ERROR:
-                        break;
-                }
-                set_error_context(get_connection(ctx_), get_result_status_name(status));
-                done(error::result_status_unexpected);
-                consume_result(get_connection(ctx_));
-            } else {
-                done();
+            set_request_result(ctx_, get_result(get_connection(ctx_)));
+
+            if (!get_request_result(ctx_)) {
+                return done();
             }
+
+            if (result_status(*get_request_result(ctx_)) != PGRES_SINGLE_TUPLE) {
+                do {
+                    while (is_busy(get_connection(ctx_))) {
+                        yield read_poll(ctx_, *this);
+                        if (auto err = consume_input(get_connection(ctx_))) {
+                            return done(err);
```

Comment:
I'm not sure that we need to propagate error here since the result has been received successfully at the moment.

Reply:
Fixed